backend/src/api/datasource/datasource.controller.spec.ts

```diff
@@ -0,0 +1,160 @@
+import { Test } from '@nestjs/testing';
+import { PrismaModule } from '../../db/prisma.module';
+import { FitnessService } from '../../integration/fitness/fitness.service';
+import { DeepMockProxy, mockDeep } from 'jest-mock-extended';
+import { MockProvider } from '../../integration/fitness/providers/mock.provider';
+import { DatasourceController } from './datasource.controller';
+import { NestRequest } from '../../types/request.type';
+import { FitnessRepository } from '../../db/repositories/fitness.repository';
+import { Response } from 'express';
+import { TestConstants } from '../../../test/lib/constants';
+import { ConsoleLogger } from '@nestjs/common';
+import { LOGGER_SERVICE } from '../../logger/logger.service';
+
+describe('Datasource controller', () => {
+  let fitnessService: DeepMockProxy<FitnessService>;
+  let fitnessController: DeepMockProxy<DatasourceController>;
+  let fitnessRepository: DeepMockProxy<FitnessRepository>;
+
+  beforeAll(async () => {
+    const module = await Test.createTestingModule({
+      imports: [PrismaModule],
+      providers: [
+        FitnessService,
+        { useClass: ConsoleLogger, provide: LOGGER_SERVICE },
+      ],
+      controllers: [DatasourceController],
+    })
+      .overrideProvider(FitnessService)
+      .useValue(mockDeep<FitnessService>())
+      .overrideProvider(FitnessRepository)
+      .useValue(mockDeep<FitnessRepository>())
+      .compile();
+
+    fitnessService = module.get(FitnessService);
+    fitnessController = module.get(DatasourceController);
+    fitnessRepository = module.get(FitnessRepository);
+  });
+
+  it('It should return the datasources for a user', async () => {
+    // Mock the datasources
+    fitnessService.getDatasourcesForUser.mockResolvedValue([
+      new MockProvider(),
+    ]);
+
+    const mockRequest = {
+      user: {
+        id: 'MOCK',
+      },
+    } as NestRequest;
```

Comment:
Well structured test, nicely done 👍. However, a lot of constants such as `mockRequest` are defined multiple times and could be extracted into a `beforeEach` or something similar.